_projects/i_gitt_scm_support.md

```diff
@@ -0,0 +1,58 @@
+---
+name: "IGitt SCM support"
+desc: "Add support for git hosters in IGitt and convert coala VCS bears to
+       use IGitt"
+requirements:
+  - "An enhancement to the coala VCS bears"
+  - "An enhancement to IGitt"
+initiatives:
+  - GSoC
+collaborating_projects:
+  - "coala"
+  - "IGitt"
+---
+
+[IGitt](https://gitlab.com/gitmate/open-source/IGitt) is a simple library
+that allows you to access various git hosting services like GitHub, GitLab
+and so on via one unified python interface.
+
+The coala bears project provides support for git and mercurial VCS but it
+does not use IGitt. With this project, VCS in coala-bears should be
+simplified using IGitt.
+
+However, there are some git hosters that are not yet supported by IGitt.
+One such hoster is BitBucket. So it is necessary to add BitBucket support to
```

Comment:
Well, there is significant difference between how BitBucket and GitHub/GitLab work. Also the REST APIs are incomplete. You can have a clean look at why we opted against it last year [here](https://hackmd.io/2YKYUYztQM-J_w6xIQu0PA).

Reply:
Can you please be little more clear on what changes are you expecting here ? @nkprince007

Reply:
Well, during GSoC 2017, it was my project to implement Bitbucket support for GitMate. It wasn't done because the REST APIs are incomplete. Bitbucket only allows `GET` requests. `PUT`, `PATCH`, `POST`, `OPTIONS` and any other methods are not supported.

Reply:
Well https://docs.atlassian.com/bitbucket-server/rest/5.11.1/bitbucket-rest.html?utm_source=%2Fstatic%2Frest%2Fbitbucket-server%2Flatest%2Fbitbucket-rest.html&utm_medium=301 shows `PUT` , `DELETE` , and `POST` options as well.

Reply:
@kriti21 Your link points to deprecated API v1.0 docs. Instead, take a look at https://developer.atlassian.com/bitbucket/api/2/reference/resource/

API v2 allows methods other than `GET`, but not for all routes. For example, there is no way to post a comment to Pull Request using API v2 (it was possible with v1, though). However, looks like most major features are supported so Bitbucket support for IGitt may be still worth implementing.

Reply:
@nkprince007 , BitBucket may be unsuitable for GitMate, but this is only about IGitt. Many parts of IGitt can be NotImplementedError for a hoster , yet still provide a useful API for building basic scripts on top of.

The critical part is whether the BitBucket support can provide what coala-bears VCS bears need, which is very little and almost certainly available via the BitBucket API.

Reply:
Worth noting that I expect Phabricator and Gerrit to also be only partial implementations, as Gerrit doesnt have many IGitt concepts at all (it doesnt have issues), and Phabricator API is a bit lacking.

Reply:
If that's the case, then it should be okay, I guess.